projects/vdk-plugins/vdk-control-api-auth/src/vdk/plugin/control_api_auth/authentication.py

```diff
@@ -101,6 +108,21 @@ def __authenticate_with_authorization_code(self):
         """
         Authenticate with authorization code as described in
         https://datatracker.ietf.org/doc/html/rfc6749#section-1.3.1
+        This type of authentication relies on pkce
+        https://datatracker.ietf.org/doc/html/rfc7636
         :return:
         """
-        raise NotImplementedError("This method is yet to be implemented.")
+        if not self._client_id or not self._auth_discovery_url:
+            print(
+                f"Login type: {self._auth_type} requires arguments: "
+                f"client-secret, client-id, auth-discovery-url"
+            )
```

Comment:
shouldn't we throw the exception in case of missing properties?

Reply:
Indeed. Changed to throw exception, instead of just printing a message. Thanks for the review